documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -2621,5 +2621,19 @@ The default cleanup mode is `ALWAYS`. You can use the
 [source,java,indent=0]
 .A test class with a temporary directory that doesn't get cleaned up
 ----
-include::{testDir}/example/TempDirCleanupModeDemo.java[tags=user_guide]
+include::{testDir}/example/TempDirectoryDemo.java[tags=user_guide_cleanup_mode]
+----
+
+`@TempDir` supports custom creation of temporary directories via the optional
+`factory` attribute.
+
+Factories can be created by implementing `TempDirFactory`. The default implementation
+available in Jupiter delegates the directory creation to
+`java.nio.file.Files::createTempDirectory`, passing `junit` as the prefix string
+to be used in generating the directory's name.
+
+[source,java,indent=0]
+.A test class with a temporary directory created by a custom factory
+----
+include::{testDir}/example/TempDirectoryDemo.java[tags=user_guide_factory]
```

Comment:
Not sure if this is enough or if there should be more. For example, should I mention `close()` for managing underlying resources like a custom `FileSystem`?

Would it make sense to add an example with an in-memory file system?

(If yes, any preference between [google/jimfs](https://github.com/google/jimfs) and [marschall/memoryfilesystem](https://github.com/marschall/memoryfilesystem)?)

Reply:
Yes, I think that would make a better example. Maybe use jimfs because it seems more popular?

Reply:
Done in d92f151.